wolvic/browser/session_settings.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WOLVIC_BROWSER_SESSION_SETTINGS_H_
+#define WOLVIC_BROWSER_SESSION_SETTINGS_H_
+
+#include <string>
+
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace content {
+
+enum class UserAgentMode {
```

Comment:
I guess the explicit values are because they need to be kept synchronized with Wolvic code.

One possibility is adding a comment stating this needs to be kept synchronized with wolvic.

But I would prefer to avoid that:
- Let's keep these without explicit values
- Let's expose in our Java API also an enum

Then, wolvic layer using SetUserAgentMode/GetUserAgentMode would do the conversion from the wolvic types to the wolvic backend of chromium side. This is basically to avoid implicit dependencies from chromium code to wolvic code.

Another nitpick. As this enum is only used for SessionSettings, I'd prefer the enum class to be declared inside the SessionSettings class. If we end up using UserAgentMode anywhere else, I'd prefer it to have its own include.

Reply:
I added a java enum to our chromium backend API and made the Wolvic side do an explicit conversion.

However, I kept the explicit values here and in SessionSettings.java since passing & converting a Java enum to the C++ side is complicated and inefficient, passing a simple int is much better in my opinion. I added comments to session_settings.h and SessionSettings.java that values have to be synchronized between these two files. But at least now there's no assumption when converting values between wolvic and Chromium backend.